sandboxes/dynamicanalysis/analyze-python.py

```diff
@@ -61,11 +66,83 @@ def importPkg(package):
         import_path = path_to_import(path)
         print('Importing', import_path)
         try:
-            importlib.import_module(import_path)
+            module = importlib.import_module(import_path)
+            execute_module(module)
         except:
             print('Failed to import', import_path)
             traceback.print_exc()
 
+
+def invoke_function(obj):
+    signature = inspect.signature(obj)
+    args = []
+    kwargs = {}
+
+    for name, param in signature.parameters.items():
+        # use MagicMock to create semi-realistic function argument values
+        # https://docs.python.org/3/library/unittest.mock.html
+        value = MagicMock() if param.default == param.empty else param.default
+
+        match param.kind:
+            case param.POSITIONAL_ONLY:
+                args.append(value)
+            case param.KEYWORD_ONLY | param.POSITIONAL_OR_KEYWORD:
+                kwargs[name] = value
+            case param.VAR_POSITIONAL:  # when *args appears in signature
+                pass  # ignore
+            case param.VAR_KEYWORD:  # when **args appears in signature
+                pass  # ignore
+
+    # bind args and invoke the function
+    # any exceptions will be propagated to the caller
+    bound = signature.bind(*args, **kwargs)
+    return obj(*bound.args, **bound.kwargs)
+
+
+def try_invoke_function(name, obj, is_method=False):
+    tag = "[method]" if is_method else "[function]"
+    print(tag, name)
+
+    try:
+        ret = invoke_function(obj)
+        print("[return value]", repr(ret))
+    except BaseException as e:
+		# catch ALL exceptions, including KeyboardInterrupt and system exit
+        print(type(e), e, sep=": ")
+
+
+def try_instantiate_class(name, obj):
```

Comment:
nit: rename to "try_instantiate_class_and_call_methods" ?

Reply:
nice, thanks